Allow disabling the pthread support check

stefano-garzarella opened the pull request:
In cases where this code is utilized only for the libraries in it (e.g. vTPM emulation in Coconut SVSM), we do not need to compile the simulator.

So it would be convenient to have the ability to disable pthread checking, which in that case is not necassary since it is used only by the simulator. Also in Coconut SVSM we don't have pthread support, so it's also a requirement and for now we have our own fork to disable this check.

By default pthread checking remains enabled, so there is no change. `--disable-pthread` is added to the configure to support this new use case.

stefano-garzarella commented:
@chrisfenner @bradlitterell @amycnelson any feedback on this?
Is there a chance we can merge it or do I close it?

We have a workaround for now, although not very nice in COCONUT SVSM: https://github.com/coconut-svsm/svsm/blob/5efb4d21b6ce03b7d06dea59b69e814f8f0a13b3/libtcgtpm/Makefile#L170

chrisfenner commented:
> @chrisfenner @bradlitterell @amycnelson any feedback on this? Is there a chance we can merge it or do I close it?
> 
> We have a workaround for now, although not very nice in COCONUT SVSM: https://github.com/coconut-svsm/svsm/blob/5efb4d21b6ce03b7d06dea59b69e814f8f0a13b3/libtcgtpm/Makefile#L170

Sorry this PR stayed dormant. It seems fine to me, so happy to approve and merge.

However, I want to share a PSA that hopefully will not be terrible news: The build system is likely to change to cmake in the next published version of this code.

stefano-garzarella commented:
> > @chrisfenner @bradlitterell @amycnelson any feedback on this? Is there a chance we can merge it or do I close it?
> > We have a workaround for now, although not very nice in COCONUT SVSM: https://github.com/coconut-svsm/svsm/blob/5efb4d21b6ce03b7d06dea59b69e814f8f0a13b3/libtcgtpm/Makefile#L170
> 
> Sorry this PR stayed dormant. It seems fine to me, so happy to approve and merge.

Great, thanks!

> 
> However, I want to share a PSA that hopefully will not be terrible news: The build system is likely to change to cmake in the next published version of this code.

Thanks for sharing. I think it's not that bad as news :-) We will need to change something, but I think we will find a way to adapt.

stefano-garzarella commented:
> Thanks for calling this out, I have re-commited your work in [bc29a21](https://github.com/TrustedComputingGroup/TPM/commit/bc29a21d44b01396223c152a4834e52318591770). I believe what happened was:
> 
>     1. We merged it as a merge commit instead of a squash or rebase.
> 
>     2. This (or some other issue) broke the cherry-picking into our internal tree, thus it got reverted by mistake when we rebased, squashed, and pushed to this public repo.
> 

Thanks for the clarification! Much appreciated!

> 
> I double checked your other recent PR (#9) and I think despite it also being a merge commit, the work was not accidentally undone.

Yep, I confirm, I did the same yesterday before commenting here, thanks for double-checking!

> 
> One more thing: we are full steam ahead on CMake now, so these files are gone in the latest internal tree. For v185 (in public review: https://trustedcomputinggroup.org/specifications-public-review/) I would say to expect the automake files to be gone. If I can be of help to your team in the transition please let me know!

I see, thanks for the update, I'll reach you if we have issue in our project (https://github.com/coconut-svsm/svsm/tree/main/libtcgtpm)
